src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/ReadyToRunCodegenCompilation.cs

```diff
@@ -70,13 +70,36 @@ public abstract class Compilation : ICompilation
 
         public bool CanInline(MethodDesc caller, MethodDesc callee)
         {
+            if (JitConfigProvider.Instance.HasFlag(CorJitFlag.CORJIT_FLAG_DEBUG_CODE))
+            {
+                // If the callee wants debuggable code, don't allow it to be inlined
+                return false;
+            }
+
+            if (callee.IsNoInlining)
+            {
+                return false;
+            }
+
             // Check to see if the method requires a security object.  This means they call demand and
             // shouldn't be inlined.
             if (callee.RequireSecObject)
             {
                 return false;
             }
 
+            // If the method is MethodImpl'd by another method within the same type, then we have
```

Comment:
Nit - I have a vague recollection this corresponds to a Crossgen1 bit; could we provide a link to the relevant code for reference?

Reply:
Do you mean a reference to the code in `CEEInfo::canInline`? On the crossgen1 side it's doing something similar, not through a bit that we set.

Reply:
No worries, just ignore this if it's not an obvious 1:1 correspondence, I'm just trying to preserve a bit of "history" when it comes to replacing Crossgen1 logic with Crossgen2.